### PR TITLE
REVIEW: host waiting interface UI

### LINKED ISSUE #35: Implement Host Interface for Quiz Application
## Objectives
- Develop a streamlined host interface for managing quiz sessions
- Create essential controls for quiz administration
- Implement basic monitoring capabilities for participant activity
- Focus on core functionality for the initial release

## Challenges
- Designing an intuitive, minimalist interface for quiz hosts
- Implementing real-time data synchronization with Firebase
- Balancing simplicity with necessary functionality
- Ensuring responsive design for different device sizes

## Completion Criteria
- Host can view and manage participants in the waiting room
- Host can initiate and control the quiz session
- Host can monitor participant progress during the quiz
- Host can view basic results and manage session lifecycle
- All interfaces work correctly across desktop and mobile devices

### 1. Host Waiting Room Implementation
- [ ] Create participant list display with real-time updates
- [ ] Implement room code display with copy functionality
- [ ] Design and implement "Start Quiz" button with validation
- [ ] Test real-time updates when participants join/leave

### 2. Host Quiz Control Panel Implementation
- [ ] Create question progress indicator (current/total questions)
- [ ] Implement countdown timer with visual indicator
- [ ] Add participant response tracker (number/percentage answered)
- [ ] Design pause/resume functionality for quiz flow
- [ ] Implement correct answer display functionality
- [ ] Test real-time synchronization of quiz state

### 3. Host Results Dashboard Implementation
- [ ] Design simple results view showing answer distribution
- [ ] Create graphical representation of option selection statistics
- [ ] Implement session end functionality
- [ ] Add new session creation option
- [ ] Test results calculation and display accuracy

### 4. Integration and Testing
- [ ] Ensure seamless transitions between host interface states
- [ ] Verify that host controls properly affect participant views
- [ ] Test real-time data synchronization throughout the quiz flow
- [ ] Perform cross-device and cross-browser compatibility testing
- [ ] Validate error handling and edge cases

### PR DESCRIPTION
**Pull request by ayakakojimaak**
## Summary
Created the Waiting Room layout for the Host side, including structural UI components.
API integration and dynamic functionality have not been implemented yet.

## Linked Issue
#35

## Changes
- Built the initial Waiting Room UI for Host
- Extracted the layout displayed with `max-heights:100vh` into a component
- Created a tab interface and made it into a component
- Created and componentized the participant list and question list

NOTE: This implementation is UI-only.
No API connection, state management, or business logic has been added at this stage.

## Screenshots:

https://github.com/user-attachments/assets/6d12372a-cbdf-4d43-920c-a5db965bc7d3

https://github.com/user-attachments/assets/cedf8b43-d0f8-4644-8f19-b2f46b059326

### CODE REVIEW

**Comment by ayakakojimaak** [comment]
@kazuki-0418 
Hi, I just sent the PR
Please check it when you have time :)

File: frontend/src/components/shared/QuestionList.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
+interface Question {
+  id: string;
+  text: string;
+  options: string[];
+  correctOption: string;
+  points: number;
+  timeLimit: number;
+  status: "waiting" | "active" | "timeout" | "completed";
+}
```
Comment (kazuki-0418) [inline frontend/src/components/shared/QuestionList.tsx]: Please store the Question type here under src/types/

File: frontend/src/components/shared/QuestionList.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+interface QuestionListProps {
```
Comment (kazuki-0418) [inline frontend/src/components/shared/QuestionList.tsx]: Can you use type instead of interface

File: frontend/src/components/ui/Tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+interface TabProps {
+  label: string;
+  isActive: boolean;
+  onClick: () => void;
+}
```
Comment (kazuki-0418) [inline frontend/src/components/ui/Tabs.tsx]: Use type instead of interface

File: frontend/src/components/ui/Tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      {label}
+    </button>
+  );
+}
+
+interface TabsProps {
+  activeTab: string;
+  onTabChange: (tab: string) => void;
+  tabs: { id: string; label: string }[];
+}
```
Comment (kazuki-0418) [inline frontend/src/components/ui/Tabs.tsx]: Use type instead of interface

File: frontend/src/components/shared/ParticipantList.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { AvatarIcon } from "@/components/ui/AvatarIcon";
+import { Badge } from "flowbite-react";
+import Image from "next/image";
+
+interface Participant {
+  id: string;
+  name: string;
+  avatar: string;
+  isGuest?: boolean;
+}
```
Comment (kazuki-0418) [inline frontend/src/components/shared/ParticipantList.tsx]: Please store the Participant type here under src/types/

File: frontend/src/components/pages/rooms/HostWaitingRoom.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+          <Image
+            src={isCopied ? "/assets/icons/check.svg" : "/assets/icons/link.svg"}
+            alt={isCopied ? "copied" : "copy"}
+            width={20}
+            height={20}
+            className={`transition-all duration-300 ${isCopied ? "text-green-500" : ""}`}
+          />
+          <span className="text-sm font-bold">Copy Link</span>
+        </button>
+      </div>
```
Comment (kazuki-0418) [inline frontend/src/components/pages/rooms/HostWaitingRoom.tsx:121]: Componentize here about URL Link Copy functionality and call it.

File: frontend/src/components/shared/Header.tsx [hunk 7]
```diff
@@ -41,73 +41,75 @@ export function Header({ username, avatarImage }: HeaderProps) {
...
-                  className="mr-3"
-                  width={20}
-                  height={20}
-                />
-                Logout
-              </button>
-            </div>
-          )}
+            {isDropdownOpen && (
+              <div className="absolute right-0 mt-2 w-40 bg-white rounded-md shadow-lg p-2 z-50 border border-gray-200">
```
Comment (kazuki-0418) [inline frontend/src/components/shared/Header.tsx]: Componentize here as AccountMenu and call it.

**Review by kazuki-0418 — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/rooms/HostWaitingRoom.tsx (hunk 6 above)
Comment (ayakakojimaak) [reply frontend/src/components/pages/rooms/HostWaitingRoom.tsx:121]: I've been working on refactoring a large component in a separate branch, so I've also separated out the URL copy functionality. I'll ask you to review it in this branch later!

WIP: participant entry UI
[frontend/src/components/shared/WaitingRoomHeader.tsx](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/57/files#diff-0eca6799072f8fcc899a914757769a3198a8f5f6579a542e91c85aed2f4131ad)

File: frontend/src/components/shared/Header.tsx (hunk 7 above)
Comment (ayakakojimaak) [reply frontend/src/components/shared/Header.tsx]: [Componentize AccountMenu](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/56/commits/3ab297287543aa4703786bda365c7088a3bd20cc)

File: frontend/src/components/shared/ParticipantList.tsx (hunk 5 above)
Comment (ayakakojimaak) [reply frontend/src/components/shared/ParticipantList.tsx]: [move types](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/56/commits/b719add5c75ee6ece99b1b925d0a9164d13a8f0b)

File: frontend/src/components/ui/Tabs.tsx (hunk 4 above)
Comment (ayakakojimaak) [reply frontend/src/components/ui/Tabs.tsx]: [move types](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/56/commits/b719add5c75ee6ece99b1b925d0a9164d13a8f0b)

File: frontend/src/components/ui/Tabs.tsx (hunk 3 above)
Comment (ayakakojimaak) [reply frontend/src/components/ui/Tabs.tsx]: [move types](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/56/commits/b719add5c75ee6ece99b1b925d0a9164d13a8f0b)

File: frontend/src/components/shared/QuestionList.tsx (hunk 2 above)
Comment (ayakakojimaak) [reply frontend/src/components/shared/QuestionList.tsx]: [move types](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/56/commits/b719add5c75ee6ece99b1b925d0a9164d13a8f0b)

File: frontend/src/components/shared/QuestionList.tsx (hunk 1 above)
Comment (ayakakojimaak) [reply frontend/src/components/shared/QuestionList.tsx]: [move types](https://github.com/Cornerstone-CICCC/nodejs-final-project-reference-error/pull/56/commits/b719add5c75ee6ece99b1b925d0a9164d13a8f0b)

**Comment by ayakakojimaak** [comment]
@kazuki-0418 
Thanks for your review! I've updated it. Could you please check if this is correct?

**Review by kazuki-0418 — APPROVED** [review]
LGTM
